video/enhanced_samples_generator.h

```diff
@@ -22,23 +22,36 @@ class EnhancedSamplesGenerator : public WaveformGenerator {
 	
 	private:
 		std::weak_ptr<audio_sample> _sample;
+
+        int previousSample = 0;
+        int currentSample = 0;
+        double samplesPerGet = 1.0;
+        double fractionalSampleOffset = 0.0;
+
+        const int baseFrequency = 16000;
```

Comment:
firstly, before I say anything else, I think this PR is really cool.  it's nice and elegantly simple.  I wish I'd thought of it when I wrote the enhanced audio stuff.  :grin:

I've been thinking about this and I'm not convinced that 16000 here is right as the "base frequency".

(for some added context, part of my thinking here is in the wider context of adding support for allowing the audio system to be reconfigured to use a higher base sample rate... as well as allowing for sample data at rates other than 16khz to be supported on the VDP.)

the underlying issue here is a mixing of the terms "frequency" and "sample rate"

16000 (actually 16384) is the base _sample rate_ of the audio system.  given that sample rate, the highest _frequency_ of note that can be conveyed in sample data is 8khz.  for a sample containing a musical note, rather than a more complex sound, the actual frequency of the note is unlikely to be 8khz, as this is _very_ high within the human hearing range, and 16khz is near (or beyond) the limit.

the upshot of this is that to play a sample at the correct/intended rate with this code as it stands you will need to play a note passing a "frequency" of 16000.  that's a very high number, far above the frequency you'd use for any of the regular waveform generators.  (16000 is _far_ outside of the range that BASIC's `SOUND` command will generate - the highest pitch value that supports of 255 results in a frequency of 4836.)

I think what may make more sense is to use a "frequency" value for playback that is more likely to match up with the actual frequency of a note contained within a sample.

the question that follows from this of course is what should that frequency be?

if I've read correctly, within mod tracker files, samples were typically sampled/tuned for C5, the C above middle-C, i.e. 523.25Hz.  this is pitch 101 from the BASIC `SOUND` command. (Acorn's BBC Micro documentation is slightly ambiguous and has a table showing it to be pitch 100, with middle C at 52, but an example immediately following the table states middle C as 53 - this latter value of 53 is how the Agon sound system is tuned)

what I'm proposing is that the "base frequency" should, by default, be 523 - i.e. that's the frequency of note that should result in 1 sample per "tick" at a 16khz sample rate.

it would make sense to allow for the base frequency of a sample to be configurable.  (precedence here in tracker mod files is they support a "fine-tune" parameter to shift the pitch of a sample up/down in eighths of a semitone.)

of course there's lots of things missing from the audio system right now to realise all of this.  several new commands are needed to configure the audio system and provide settings parameters for samples.  (beyond parameters for frequency and sample rate, as discussed above, looping info should also get added - and will also needed for tracker file support.)  I'm working on it :grin: